### PR TITLE
Make RefCountingRunnable.AcquiredRef lighter

### PR DESCRIPTION
**Pull request by original-brownbear**
Just a small improvement saving some allocations when this may be hot. Found from benchmarking blob cache.

### CODE REVIEW

**Review by thecoop — APPROVED** [review]

**Review by DaveCTurner — CHANGES_REQUESTED** [review]
`AcquiredRef` is really only there for assertions, and for some (subjective) clarity of the code. Since this is showing up on hot paths I think we can sacrifice the clarity aspect, and so we can drop `AcquiredRef` and return `Releasables#assertOnce(this)` from each `acquire` call.

**Review by DaveCTurner — APPROVED** [review]
LGTM

File: server/src/main/java/org/elasticsearch/action/support/RefCountingRunnable.java [hunk 1]
```diff
@@ -103,7 +87,7 @@ public RefCountingRunnable(Runnable delegate) {
      */
     public Releasable acquire() {
         if (refCounted.tryIncRef()) {
-            return new AcquiredRef();
+            return Releasables.assertOnce(this);
```
Comment (DaveCTurner) [inline server/src/main/java/org/elasticsearch/action/support/RefCountingRunnable.java:90]: Probably deserves a comment tho:

```suggestion
            // All refs are considered equal so there's no real need to allocate a new object here, although note that this deviates
            // (subtly) with the docs for Closeable#close() which indicate that it should be idempotent. But only if assertions are
            // disabled, and if assertions are enabled then we are asserting that we never double-close these things anyway.
            return Releasables.assertOnce(this);
```

**Review by henningandersen — APPROVED** [review]
LGTM2.

**Comment by original-brownbear** [comment]
Thanks all!

**Pull request merged**